test/sqllogictest/attributes/mir_arity.slt

```diff
@@ -96,9 +96,14 @@ EXPLAIN OPTIMIZED PLAN WITH(arity) AS TEXT FOR
 SELECT sum(e * f), max(f) FROM v GROUP BY mod(e, 5)
 ----
 Explained Query:
-  Project (#1, #2) // { arity: 2 }
-    Reduce group_by=[(#0 % 5)] aggregates=[sum((#0 * #1)), max(#1)] // { arity: 3 }
-      ReadStorage materialize.public.v // { arity: 2 }
+  Project (#1, #3) // { arity: 2 }
+    Join on=(#0 = #2) type=differential // { arity: 4 }
+      ArrangeBy keys=[[#0]] // { arity: 2 }
+        Reduce group_by=[(#0 % 5)] aggregates=[sum((#0 * #1))] // { arity: 2 }
+          ReadStorage materialize.public.v // { arity: 2 }
+      ArrangeBy keys=[[#0]] // { arity: 2 }
+        Reduce group_by=[(#0 % 5)] aggregates=[max(#1)] // { arity: 2 }
```

Comment:
Did the old version compute the key expression only once? If yes, then this is a possibility for a CPU regression.

We could prevent this by putting a complex key expression in a Map before each of the Reduces, in which case RelationCSE would grab also that Map.

Reply:
Probably! Though I think you could go either way on regression or not. Now, rather than doing all the keys and values for each reduction path, and then forking those into each reduction path, we "potentially" use less memory by creating the actual inputs directly, rather than sharing the work and then needing to clone the data to project it down.

Reply:
Ok, then this is fine. In any case, CPU regressions are much less scary than memory regressions.